lib/src/util/console_log_utils.dart

```diff
@@ -14,57 +14,89 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import 'dart:async';
 import 'dart:js';
 
 import 'package:meta/meta.dart';
 import 'package:react/react_client/react_interop.dart';
 
-/// Runs a provided callback and returns the logs that occur during the runtime
+/// Runs a provided [callback] and [print]s each log that occurs during the runtime
 /// of that function.
 ///
-/// Can be used to capture logs, warnings, or errors as specified by setting
-/// the [configuration]. To set the [configuration], pass in the corresponding
-/// config class ([ConsoleConfig.log], [ConsoleConfig.warn], [ConsoleConfig.error], [ConsoleConfig.all]).
+/// [print] is called in the same Zone as this function was called in, similar to a Stream subscription.
+///
+/// Can be used to print logs, warnings, or errors based on [configuration].
 ///
 /// The function assumes that any `propType` warnings that occur during
 /// the function runtime should be captured. Consequently, the `PropType` cache
 /// is reset prior to calling the provided callback.
+T printConsoleLogs<T>(
+  T Function() callback, {
+  ConsoleConfig configuration = ConsoleConfig.all,
+}) => spyOnConsoleLogs(callback, configuration: configuration, onLog: print);
+
+/// Runs a provided [callback] and calls [onLog] for each log that occurs during the runtime
+/// of that function.
+///
+/// [onLog] is called in the same Zone as this function was called in, similar to a Stream subscription.
+///
+/// Can be used to capture logs, warnings, or errors based on [configuration].
+///
+/// The function assumes that any `propType` warnings that occur during
+/// the function runtime should be captured. Consequently, the `PropType` cache
+/// is reset prior to calling the provided callback.
+T spyOnConsoleLogs<T>(
+  T Function() callback, {
+  @required void Function(String) onLog,
+  ConsoleConfig configuration = ConsoleConfig.all,
+}) {
+  final stopSpying = startSpyingOnConsoleLogs(configuration: configuration, onLog: onLog);
+  try {
+    return callback();
+  } finally {
+    stopSpying();
+  }
+}
+
+/// Starts spying on concole logs, calling [onLog] for each log that occurs until the
+/// returned function (`stopSpying`) is called.
 ///
-/// If any errors are thrown during the callback, e.g. during a render that expects
-/// props that are not valid, the errors will be caught to allow the test to complete.
-List<String> recordConsoleLogs(
```

Comment:
`recordConsoleLogs` still exists, but was moved to `test/unit/util/prints_and_logs_recording.dart`